pkg/sql/sqlstats/insights/ingester.go

```diff
@@ -68,6 +68,8 @@ var eventBufferPool = sync.Pool{
 	New: func() interface{} { return new(eventBuffer) },
 }
 
+// event is a single event that can be observed by the ingester.
+// Only one of the fields will be non-nil.
```

Comment:
nit: I think this is a little confusing because its possible for both to be nil. Maybe something like: "At most, one of transaction or statement may be non-nil." would be more clear?

Reply:
Done.